x/metrics.go

```diff
@@ -17,256 +17,277 @@
-		"dgraph_memory_idle_bytes": prometheus.NewDesc(
-			"dgraph_memory_idle_bytes",
-			"dgraph_memory_idle_bytes",
-			nil, nil,
-		),
-		"dgraph_memory_proc_bytes": prometheus.NewDesc(
-			"dgraph_memory_proc_bytes",
-			"dgraph_memory_proc_bytes",
-			nil, nil,
-		),
-		"dgraph_active_mutations_total": prometheus.NewDesc(
-			"dgraph_active_mutations_total",
-			"dgraph_active_mutations_total",
-			nil, nil,
-		),
-		"dgraph_predicate_stats": prometheus.NewDesc(
-			"dgraph_predicate_stats",
-			"dgraph_predicate_stats",
-			[]string{"name"}, nil,
-		),
-		"badger_disk_reads_total": prometheus.NewDesc(
-			"badger_disk_reads_total",
-			"badger_disk_reads_total",
-			nil, nil,
-		),
-		"badger_disk_writes_total": prometheus.NewDesc(
-			"badger_disk_writes_total",
-			"badger_disk_writes_total",
-			nil, nil,
-		),
-		"badger_read_bytes": prometheus.NewDesc(
-			"badger_read_bytes",
-			"badger_read_bytes",
-			nil, nil,
-		),
-		"badger_written_bytes": prometheus.NewDesc(
-			"badger_written_bytes",
-			"badger_written_bytes",
-			nil, nil,
-		),
-		"badger_lsm_level_gets_total": prometheus.NewDesc(
-			"badger_lsm_level_gets_total",
-			"badger_lsm_level_gets_total",
-			[]string{"level"}, nil,
-		),
-		"badger_lsm_bloom_hits_total": prometheus.NewDesc(
-			"badger_lsm_bloom_hits_total",
-			"badger_lsm_bloom_hits_total",
-			[]string{"level"}, nil,
-		),
-		"badger_gets_total": prometheus.NewDesc(
-			"badger_gets_total",
-			"badger_gets_total",
-			nil, nil,
-		),
-		"badger_puts_total": prometheus.NewDesc(
-			"badger_puts_total",
-			"badger_puts_total",
-			nil, nil,
-		),
-		"badger_memtable_gets_total": prometheus.NewDesc(
-			"badger_memtable_gets_total",
-			"badger_memtable_gets_total",
-			nil, nil,
-		),
-		"badger_lsm_size": prometheus.NewDesc(
-			"badger_lsm_size",
-			"badger_lsm_size",
-			[]string{"dir"}, nil,
-		),
-		"badger_vlog_size": prometheus.NewDesc(
-			"badger_vlog_size",
-			"badger_vlog_size",
-			[]string{"dir"}, nil,
-		),
+	pe, err := prometheus.NewExporter(prometheus.Options{
+		Namespace: "dgraph", // TODO: read this namespace from flags
+		// TODO: Enable an on error that logs to Dgraph's logging output.
 	})
-	prometheus.MustRegister(expvarCollector)
-	http.Handle("/debug/prometheus_metrics", prometheus.Handler())
+	if err != nil {
+		log.Fatalf("Failed to create OpenCensus Prometheus exporter: %v", err)
+	}
+
+	http.Handle("/debug/prometheus_metrics", pe)
+}
+
+// ObservabilityEnabledParentContext returns a context with tags that are useful for
+// distinguishing the state of the running system. It contains tags such as:
+// * PID
+// * OS
+// * Architecture
+// This context will be used to derive other contexts.
+func ObservabilityEnabledParentContext() context.Context {
+	// At the beginning add some distinguishing information
+	// to the context as tags that will be propagated when
+	// collecting metrics.
+	ctx, _ := tag.New(context.Background(),
+		tag.Upsert(KeyPid, fmt.Sprintf("%d", os.Getpid())),
```

Comment:
Don't think we need all this. We can keep our contexts cheap by only setting whatever is absolutely necessary.